x-pack/legacy/plugins/snapshot_restore/public/app/components/policy_delete_provider.tsx

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, useRef, useState } from 'react';
+import { EuiConfirmModal, EuiOverlayMask } from '@elastic/eui';
+import { useAppDependencies } from '../index';
+import { deletePolicies } from '../services/http';
+
+interface Props {
+  children: (deletePolicy: DeletePolicy) => React.ReactElement;
+}
+
+export type DeletePolicy = (names: string[], onSuccess?: OnSuccessCallback) => void;
+
+type OnSuccessCallback = (policiesDeleted: string[]) => void;
+
+export const PolicyDeleteProvider: React.FunctionComponent<Props> = ({ children }) => {
+  const {
+    core: {
+      i18n,
+      notification: { toastNotifications },
+    },
+  } = useAppDependencies();
+  const { FormattedMessage } = i18n;
+  const [policyNames, setPolicyNames] = useState<string[]>([]);
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const onSuccessCallback = useRef<OnSuccessCallback | null>(null);
+
+  const deletePolicyPrompt: DeletePolicy = (names, onSuccess = () => undefined) => {
+    if (!names || !names.length) {
+      throw new Error('No policy names specified for deletion');
+    }
+    setIsModalOpen(true);
+    setPolicyNames(names);
+    onSuccessCallback.current = onSuccess;
+  };
+
+  const closeModal = () => {
+    setIsModalOpen(false);
+    setPolicyNames([]);
+  };
+
+  const deletePolicy = () => {
+    const policiesToDelete = [...policyNames];
+    deletePolicies(policiesToDelete).then(({ data, error }) => {
+      const { itemsDeleted, errors } = data || { itemsDeleted: undefined, errors: undefined };
```

Comment:
Nit: we could also assign a default value to the `data` parameter above, which would be a bit terser. We could also set a default value to `itemsDeleted` and `errors`, which would make lines 52 and 72 terser as well.

```js
    deletePolicies(policiesToDelete).then(({ data = {}, error }) => {
      const { itemsDeleted = [], errors = [] } = data;
```

```js
// ln 52
      if (itemsDeleted.length) {
```

```js
// ln 72
      if (error || errors.length) {
```

I just realized that this won't quite work with the current logic for displaying error messages, but I have a suggestion that could make it work in the next comment.